lib/bin/node-spellchecker/build-linux-debian.sh

```diff
@@ -1,39 +1,50 @@
 #!/bin/bash
 
-# requires node-gyp (npm install -g node-gyp) plus CLI developper tools
-
-node-gyp rebuild --target=6.0.9 --arch=ia32 --dist-url=https://atom.io/download/electron
-cp build/Release/spellchecker.node ../spellchecker-deb-6.0.9-ia32.node
-
-node-gyp rebuild --target=6.0.9 --arch=x64 --dist-url=https://atom.io/download/electron
-cp build/Release/spellchecker.node ../spellchecker-deb-6.0.9-x64.node
-
-node-gyp rebuild --target=7.1.7 --arch=ia32 --dist-url=https://atom.io/download/electron
-cp build/Release/spellchecker.node ../spellchecker-deb-7.1.7-ia32.node
-
-node-gyp rebuild --target=7.1.7 --arch=x64 --dist-url=https://atom.io/download/electron
-cp build/Release/spellchecker.node ../spellchecker-deb-7.1.7-x64.node
-
-node-gyp rebuild --target=9.1.0 --arch=ia32 --dist-url=https://atom.io/download/electron
-cp build/Release/spellchecker.node ../spellchecker-deb-9.1.0-ia32.node
-
-node-gyp rebuild --target=9.1.0 --arch=x64 --dist-url=https://atom.io/download/electron
-cp build/Release/spellchecker.node ../spellchecker-deb-9.1.0-x64.node
-
-node-gyp rebuild --target=11.1.0 --arch=ia32 --dist-url=https://atom.io/download/electron
-cp build/Release/spellchecker.node ../spellchecker-deb-11.1.0-ia32.node
-
-node-gyp rebuild --target=11.1.0 --arch=x64 --dist-url=https://atom.io/download/electron
-cp build/Release/spellchecker.node ../spellchecker-deb-11.1.0-x64.node
-
-node-gyp rebuild --target=12.0.4 --arch=ia32 --dist-url=https://atom.io/download/electron
-cp build/Release/spellchecker.node ../spellchecker-deb-12.0.4-ia32.node
-
-node-gyp rebuild --target=12.0.4 --arch=x64 --dist-url=https://atom.io/download/electron
-cp build/Release/spellchecker.node ../spellchecker-deb-12.0.4-x64.node
-
-node-gyp rebuild --target=13.1.7 --arch=ia32 --dist-url=https://atom.io/download/electron
-cp build/Release/spellchecker.node ../spellchecker-deb-13.1.7-ia32.node
-
-node-gyp rebuild --target=13.1.7 --arch=x64 --dist-url=https://atom.io/download/electron
-cp build/Release/spellchecker.node ../spellchecker-deb-13.1.7-x64.node
+# requires node-gyp (installed as a package.json dependency) plus CLI developer tools   
+# target versions are nodejs versions, not electron versions. 
+# The versions used in the past need to be grabbed from VSCode release notes and / or the git history at
+# https://github.com/microsoft/vscode/blob/main/cgmanifest.json#L55 
+# and / or https://github.com/microsoft/vscode/blob/main/remote/.yarnrc 
+# 16.5.0 (the nodejs version in electron 15, not yet in VSCode, but expected to be used then)
```

Comment:
My main question still to be clarified is about the choice of nodejs versions, why the specific versions used were looking rather like electron versions and why the nodejs headers were pulled from the atom servers instead of the defaults.   It could be that this is some gross misunderstanding but adjusting the version choice back is trivial. 

This is now building binaries for the nodejs version used in the VSCode versions I could find, but only for the latest of a given nodejs major version.

Reply:
I do not know enough about nodejs, to provide any comments on this. I was just always sure that I was following Electron's version to match the binaries. Not nodejs's.

Reply:
Thanks for the hint, I found https://www.electronjs.org/docs/tutorial/using-native-node-modules now,  they explicitly state that the electron bundled nodejs is not 100% binary compatible with "plain" node.   They have a new URL for the headers ( `--dist-url=https://electronjs.org/headers` ), but that's just a migration they did some time ago ( https://github.com/electron/electron/commit/e73a0e6cc66665d83242e263bc8718c42ffd7c7e ) 

From that page I found https://github.com/prebuild/prebuild  which looks like it bundles what I'm trying to achieve here in a better maintained way. It did not work OOTB so for now I'll dial it all back to your proven selection of versions.